test/test_buffer.cpp

```diff
@@ -350,6 +352,156 @@ namespace sparrow
+                CHECK_EQ(b[1], expected_value);
+                CHECK_EQ(b[2], expected_value);
+                CHECK_EQ(b[3], 0);
+                CHECK_EQ(b[4], 1);
+                CHECK_EQ(b[5], 2);
+                CHECK_EQ(b[6], 3);
+            }
+
+            SUBCASE("Inserts elements from range [first, last) before pos")
+            {
+                constexpr std::size_t size = 4u;
+                buffer_test_type b(make_test_buffer(size), size);
+
+                const std::vector<uint8_t> values = {101, 102, 103};
+                const std::size_t expected_new_size = size + values.size();
+                b.insert(b.cbegin(), values.cbegin(), values.cend());
+                const std::size_t new_size = b.size();
+                REQUIRE_EQ(new_size, expected_new_size);
+                CHECK_EQ(b[0], 101);
+                CHECK_EQ(b[1], 102);
+                CHECK_EQ(b[2], 103);
+                CHECK_EQ(b[3], 0);
+                CHECK_EQ(b[4], 1);
+                CHECK_EQ(b[5], 2);
+                CHECK_EQ(b[6], 3);
+            }
+
+            SUBCASE("Inserts elements from initializer list before pos")
+            {
+                constexpr std::size_t size = 4u;
+                buffer_test_type b(make_test_buffer(size), size);
+
+                b.insert(b.cbegin(), {101, 102, 103});
+                REQUIRE_EQ(b.size(), size + 3);
+                CHECK_EQ(b[0], 101);
+                CHECK_EQ(b[1], 102);
+                CHECK_EQ(b[2], 103);
+                CHECK_EQ(b[3], 0);
+                CHECK_EQ(b[4], 1);
+                CHECK_EQ(b[5], 2);
+                CHECK_EQ(b[6], 3);
+            }
+        }
+
+        TEST_CASE("erase")
+        {
+            SUBCASE("Removes the element at pos")
+            {
+                constexpr std::size_t size = 4u;
+                buffer_test_type b(make_test_buffer(size), size);
+
+                b.erase(b.cbegin());
+                REQUIRE_EQ(b.size(), size - 1);
+                CHECK_EQ(b[0], 1);
+                CHECK_EQ(b[1], 2);
+                CHECK_EQ(b[2], 3);
+            }
+
+            SUBCASE("Removes the elements in the range [first, last)")
+            {
+                constexpr std::size_t size = 4u;
+                buffer_test_type b(make_test_buffer(size), size);
+
+                b.erase(b.cbegin(), b.cbegin() + 2);
+                REQUIRE_EQ(b.size(), size - 2);
+                CHECK_EQ(b[0], 2);
+                CHECK_EQ(b[1], 3);
+            }
+        }
+
+        TEST_CASE("push_back")
+        {
+            SUBCASE("The new element is initialized as a copy of value")
+            {
+                constexpr std::size_t size = 4u;
+                buffer_test_type b(make_test_buffer(size), size);
+
+                constexpr uint8_t expected_value = 101;
+                b.push_back(expected_value);
+                REQUIRE_EQ(b.size(), size + 1);
+                CHECK_EQ(b[size], expected_value);
```

Comment:
Add a check that `end()` points to the element after the added element. Same for all `push_back` tests.